cni/network/network.go

```diff
@@ -421,27 +423,37 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 		telemetry.SendCNIMetric(&cniMetric, plugin.tb)
 
 		// Add Interfaces to result.
-		// previously just logged the default (infra) interface so this is equivalent behavior
+		// previously we had a default interface info to select which interface info was the one to be returned from cni add
 		cniResult := &cniTypesCurr.Result{}
 		for key := range ipamAddResult.interfaceInfo {
-			logger.Info("Exiting add, interface info retrieved", zap.Any("ifInfo", ipamAddResult.interfaceInfo[key]))
-			// previously we had a default interface info to select which interface info was the one to be returned from cni add
 			// now we have to infer which interface info should be returned
 			// we assume that we want to return the infra nic always, and if that is not found, return any one of the secondary interfaces
 			// if there is an infra nic + secondary, we will always return the infra nic (linux swift v2)
-			cniResult = convertInterfaceInfoToCniResult(ipamAddResult.interfaceInfo[key], args.IfName)
+			cniResult = plugin.convertInterfaceInfoToCniResult(ipamAddResult.interfaceInfo[key], args.IfName)
 			if ipamAddResult.interfaceInfo[key].NICType == cns.InfraNIC {
 				break
 			}
 		}
 
-		addSnatInterface(nwCfg, cniResult)
+		// stdout multiple cniResults for containerd to create multiple pods
+		// containerd receives each cniResult as the stdout and create pod
+		addSnatInterface(nwCfg, cniResult) //nolint TODO: check whether Linux supports adding secondary snatinterface
+
+		// add IB NIC interfaceInfo to cniResult
+		for _, epInfo := range epInfos {
+			if epInfo.NICType == cns.BackendNIC {
+				cniResult = plugin.addIBInterfaceInfoToCniResult(cniResult, epInfo)
+				break
+			}
+		}
+
+		logger.Info("CNIResult", zap.Any("Add Call()", cniResult))
```

Comment:
can we also log error field in same message?

Reply:
this is a debug message, let's not print this cniResult...